### DISCUSSION TITLE
How to make plugins for deno fmt

### DISCUSSION THREAD

**Discussion by daalfox** [denoland/deno#21934] [Q&A] [answered]
Can I make plugins for deno fmt to extend it? e.g. for supporting class sorting and attribute sorting in tsx files etc.
If not, is there an issue tracking this feature?

**Comment by nberlette** [comment, marked as answer]
Sorry for digging up an old thread, but I felt the need to chime in.

See [dprint](https://github.com/dprint/dprint) and [dprint-plugin-typescript](https://github.com/dprint/dprint-plugin-typescript). It's the formatter Deno uses under the hood, created by absolute unit @dsherret. It's written in Rust, supports multiple runtimes with a WASM build, among other things. 

While `deno fmt` doesn't offer much in terms of configurability, `dprint` has configurability like no other. You can dial in its behavior with practically every type of change it applies. That granularity of control (and probably its performance, too) were partially responsible for it becoming the formatter of choice for TypeScript itself not too long ago. 

FYI: You can simplify the configuration process with the `deno: true` option in `dprint.json`. That pre-populates the rest of the configurations to the values that `deno fmt` uses. You can then override individual options as needed.

**Reply by daalfox** [reply]
That's great to know. Thank you for your help!
